grafana/grafana
# Dashboard: Fix Regression detected in time range variables under the refactorVariablesTimeRange feature flag

**What is this feature?**
The original algorithm did not account for situations where a variable set to refresh based on time range had a dependency on another variable that was not configured to refresh on time range.

##### The original dashboard from the customer:
A query variable with dependent variables relied on a DataSource variable. However, currently, we don't allow DataSource variables to be configured to refresh based on time range. In such cases, we need to add that original query variable to the list of variables to be refreshed.

![variables system_9~2](https://github.com/grafana/grafana/assets/239999/cece0c43-9925-49b3-aee6-b43c29f9ae25)

**How to test this PR?**

- Enable locally `refactorVariablesTimeRange` feature flag
- Configure Prometheus datasource (use [robust perception](http://demo.robustperception.io:9090/))
- Import this dashboard: [Slow Template Variable Basic Exampless-1693929458840.txt](https://github.com/grafana/grafana/files/12525781/Slow.Template.Variable.Basic.Exampless-1693929458840.txt)
- Modify the time range of the dashboard
- Check the network and verify that 3 calls are being made 


**Why do we need this feature?**

Will fix slow template variable issue

**Who is this feature for?**

users with the feature flag enabled for now and later everyone 

**Which issue(s) does this PR fix?**:



Fixes #71060

**Special notes for your reviewer:**

- [x] Missing unit test of edge case
- [x] Fix betterer
- [x] Add better description and example to the PR

Please check that:
